--- gnutls-3.4.0/doc/manpages/Makefile.am.orig	2015-04-06 10:48:30.000000000 +0200
+++ gnutls-3.4.0/doc/manpages/Makefile.am	2015-04-19 17:23:52.212283722 +0200
@@ -134,11 +134,8 @@
 APIMANS += gnutls_certificate_get_peers_subkey_id.3
 APIMANS += gnutls_certificate_get_trust_list.3
 APIMANS += gnutls_certificate_get_verify_flags.3
-APIMANS += gnutls_certificate_get_verify_flags.3
-APIMANS += gnutls_certificate_get_x509_crt.3
 APIMANS += gnutls_certificate_get_x509_crt.3
 APIMANS += gnutls_certificate_get_x509_key.3
-APIMANS += gnutls_certificate_get_x509_key.3
 APIMANS += gnutls_certificate_send_x509_rdn_sequence.3
 APIMANS += gnutls_certificate_server_set_request.3
 APIMANS += gnutls_certificate_set_dh_params.3
