--- gnutls-2.2.1/libextra/Makefile.am.orig	2008-01-17 18:45:55.000000000 +0100
+++ gnutls-2.2.1/libextra/Makefile.am	2008-01-21 21:15:31.324404586 +0100
@@ -60,7 +60,7 @@
 
 libgnutls_extra_la_SOURCES = gnutls_extra.c
 
-libgnutls_openssl_la_LDFLAGS = -no-undefined -L$(top_srcdir)/lib/.libs
+libgnutls_openssl_la_LDFLAGS = -no-undefined
 
 # OpenSSL
 
@@ -90,7 +90,7 @@
 # OpenPGP
 
 libgnutls_extra_la_LIBADD =
-libgnutls_extra_la_LDFLAGS = -no-undefined -L$(top_srcdir)/lib/.libs
+libgnutls_extra_la_LDFLAGS = -no-undefined
 if ENABLE_OPENPGP
 libgnutls_extra_la_SOURCES += gnutls_openpgp.c
 libgnutls_extra_la_LIBADD += openpgp/libgnutls_openpgp.la
