--- gnutls-2.12.0/lib/m4/hooks.m4.orig	2011-03-10 16:54:36.000000000 +0100
+++ gnutls-2.12.0/lib/m4/hooks.m4	2011-04-02 11:22:10.510496900 +0200
@@ -111,7 +111,7 @@
   if test "$included_pakchois" = "no"; then
     AC_LIB_HAVE_LINKFLAGS(pakchois,, [#include <pakchois/pakchois.h>],
                           [pakchois_module_load(0,0);])
-    if test "$ac_cv_pakchois" != yes; then
+    if test "$ac_cv_libpakchois" != yes; then
       included_pakchois=yes
       AC_MSG_WARN([[
   *** 
