--- gnutls-2.8.5/lib/m4/lib-link.m4.orig	2009-11-02 11:34:50.000000000 +0100
+++ gnutls-2.8.5/lib/m4/lib-link.m4	2010-01-31 19:56:56.126778128 +0100
@@ -322,7 +322,11 @@
           fi
           if test "X$found_dir" != "X"; then
             dnl Found the library.
+            if test "X$found_dir" = "X/usr/$acl_libdirstem"; then
+                LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }-l$name"
+	    else
             LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }-L$found_dir -l$name"
+	    fi
             if test "X$found_so" != "X"; then
               dnl Linking with a shared library. We attempt to hardcode its
               dnl directory into the executable's runpath, unless it's the
