--- gnutls-3.0.13/m4/lib-link.m4.orig	2008-05-17 07:44:04.000000000 +0200
+++ gnutls-3.0.13/m4/lib-link.m4	2011-04-02 11:29:18.159546509 +0200
@@ -322,7 +322,11 @@
           fi
           if test "X$found_dir" != "X"; then
             dnl Found the library.
+            if test "X$found_dir" = "X/usr/$acl_libdirstem"; then
+                LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }-l$name"
+	    else
             LTLIB[]NAME="${LTLIB[]NAME}${LTLIB[]NAME:+ }-L$found_dir -l$name"
+	    fi
             if test "X$found_so" != "X"; then
               dnl Linking with a shared library. We attempt to hardcode its
               dnl directory into the executable's runpath, unless it's the
