--- gnutls-3.1.6/configure.ac.orig	2013-01-02 21:14:55.491355416 +0100
+++ gnutls-3.1.6/configure.ac	2013-01-02 21:23:27.598011407 +0100
@@ -27,7 +27,7 @@
 
 AM_INIT_AUTOMAKE([1.11.3 no-dist-gzip dist-xz dist-lzip -Wall  -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_MSG_RESULT([***
 *** Checking for compilation programs...
