--- gnutls-0.8.11/libextra/gnutls_openpgp.c.orig	Sat Oct  4 08:39:41 2003
+++ gnutls-0.8.11/libextra/gnutls_openpgp.c	Wed Oct  8 13:23:12 2003
@@ -510,7 +510,7 @@
                         key_attr_t by, opaque *pattern )
 {
     keybox_blob *blob = NULL;
-    CDK_KEYDB_HD hd = NULL;
+    cdk_keydb_hd_t hd = NULL;
     cdk_kbnode_t knode = NULL;
     unsigned long keyid[2];
     unsigned char *buf;
@@ -1185,7 +1185,7 @@
                            int cert_list_length )
 {
     cdk_kbnode_t knode = NULL;
-    CDK_KEYDB_HD hd = NULL;
+    cdk_keydb_hd_t hd = NULL;
     keybox_blob *blob = NULL;
     int rc = 0;
     int status = 0;
